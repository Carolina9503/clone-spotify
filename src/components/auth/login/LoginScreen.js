import React from "react";
import { loginURL } from "./logicLogin";
import styles from "./LoginScreen.module.scss";

const LoginScreen = () => {
  return (
    <div className={styles.container}>
      <img
        src="http://1000logos.net/wp-content/uploads/2017/08/Spotify-symbol.jpg"
        alt="Logo Spotify"
        className={styles.logo}
      ></img>
      <a className={styles.container_a} href={loginURL}>
        Login with spotify
      </a>
    </div>
  );
};

export default LoginScreen;
