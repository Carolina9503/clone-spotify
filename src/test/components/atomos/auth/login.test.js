import React from "react";
import { shallow } from "enzyme";
import LoginScreen from "../../../../components/atomos/auth/login/LoginScreen";

describe("Pruebas en <LoginScreen/>", () => {
  const wrapper = shallow(<LoginScreen />);

  test("debe de mostar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener una <a></a> con su texto", () => {
    const a = wrapper.find("a");
    const texto = "Login with spotify";
    expect(a.text().trim()).toBe(texto);
  });

  test("debe de tener una imagen con un ulr y alt", () => {
    const url =
      "http://1000logos.net/wp-content/uploads/2017/08/Spotify-symbol.jpg";
    const img = wrapper.find("img");
    const alt = "Logo Spotify";
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(alt);
  });

  test("debe tener la clase container", () => {
    const div = wrapper.find("div");
    expect(div.prop("className")).toBe("container");
  });
});
