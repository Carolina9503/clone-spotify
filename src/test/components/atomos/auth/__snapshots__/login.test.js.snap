// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe de mostar el componente correctamente 1`] = `
<div
  className="container"
>
  <img
    alt="Logo Spotify"
    className="logo"
    src="http://1000logos.net/wp-content/uploads/2017/08/Spotify-symbol.jpg"
  />
  <a
    className="container_a"
    href="https://accounts.spotify.com/authorize?client_id=6ce78825323c48188d029e27ff16c54d&response_type=token&redirect_uri=http://localhost:3000/&scope=user-read-currently-playing%20user-read-recently-played%20user-read-playback-state%20user-top-read%20user-modify-playback-state&show_dialog=true"
  >
    Login with spotify
  </a>
</div>
`;

exports[`Pruebas en <LoginScreen/> debe de mostarr el componente correctamente 1`] = `
<div
  className="container"
>
  <img
    alt="Logo Spotify"
    className="logo"
    src="http://1000logos.net/wp-content/uploads/2017/08/Spotify-symbol.jpg"
  />
  <a
    className="container_a"
    href="https://accounts.spotify.com/authorize?client_id=6ce78825323c48188d029e27ff16c54d&response_type=token&redirect_uri=http://localhost:3000/&scope=user-read-currently-playing%20user-read-recently-played%20user-read-playback-state%20user-top-read%20user-modify-playback-state&show_dialog=true"
  >
    Login with spotify
  </a>
</div>
`;
